### PR TITLE
Added alertDialog for reserving cards from deck

### PR DESCRIPTION
**Pull request by KamilDevelopments**
Added alertDialog for reserving cards from deck + also added uuid to test cards that someone missed, while refactoring Card.java

### CODE REVIEW

File: app/src/main/java/com/example/splendormobilegame/activities/GameActivity/GameActivity.java [hunk 1]
```diff
@@ -312,13 +316,59 @@ public void onClick(DialogInterface dialog, int which) {
...
+                })
+                .show();
+    }
+
+    private void setupReservingFromDeckButtons() {
+        binding.CardPile1CardView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                //reserve 1 tier card
+                showDeckReserveDialog(1);
```
Comment (kestivvi) [inline app/src/main/java/com/example/splendormobilegame/activities/GameActivity/GameActivity.java:345]: You have corrected this to provide number as a card tier, but `showDeckReserveDialog` still expects UUID as parameter.

File: app/src/main/java/com/example/splendormobilegame/activities/GameActivity/GameActivity.java [hunk 2]
```diff
@@ -312,13 +316,59 @@ public void onClick(DialogInterface dialog, int which) {
                 })
                 .show();
     }
-    public void setupCardsFirstTierRecyclerView(){
+
+    public void showDeckReserveDialog(UUID cardUuid) {
```
Comment (kestivvi) [inline app/src/main/java/com/example/splendormobilegame/activities/GameActivity/GameActivity.java]: This still expects UUID as a parameter, but should expect int instead.

**Review by kestivvi — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/splendormobilegame/activities/GameActivity/GameActivity.java (hunk 2 above)
Comment (KamilDevelopments) [reply app/src/main/java/com/example/splendormobilegame/activities/GameActivity/GameActivity.java]: yup, I missed that

**Review by kestivvi — APPROVED** [review]
Great 🎉🎂🪁🪁

**Pull request merged**